src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,251 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query
+                = "INSERT INTO cars (model, manufacturers_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement
+                         = connection.prepareStatement(query,
+                         Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.executeUpdate();
+
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                Long carId = resultSet.getObject(1, Long.class);
+                car.setId(carId);
+            }
+            addDriversToCar(car, connection);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create car to DB. For car: " + car, e);
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query
+                = "SELECT * FROM cars AS c "
+                + "LEFT JOIN manufacturers AS m ON "
+                + "c.manufacturers_id = m.id "
+                + "WHERE c.id = ? AND c.is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement
+                        = connection.prepareStatement(query)) {
+
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            Car car = null;
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+                car.setDrivers(getDrivers(connection, car.getId()));
+            }
+            return Optional.ofNullable(car);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error retrieving car " + "by id = " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query
+                = "SELECT * FROM cars AS c "
```

Comment:
It's better to use 'JOIN' instead of 'LEFT JOIN' in this query to avoid retrieving cars without manufacturers. 🚗

If you expect returning cars without manufacturers, keep it as is